Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ButtonGroup.cpp

```diff
@@ -31,14 +31,19 @@ namespace AzToolsFramework::ViewportUi::Internal
     {
         if (auto buttonEntry = m_buttons.find(buttonId); buttonEntry != m_buttons.end())
         {
-            for (auto& button : m_buttons)
-            {
-                button.second->m_state = Button::State::Deselected;
-            }
+            ClearHighlightedButton();
             buttonEntry->second->m_state = Button::State::Selected;
         }
     }
 
+    void ButtonGroup::ClearHighlightedButton()
```

Comment:
update: Should this be called `ClearHighlightedButtons` (plural) as we're looping through all of them?

Reply:
Or maybe we could improve this by caching the 'currently' highlighted button that we then clear?

Reply:
i'll cache the last highlighted button, that makes more sense to me.

Reply:
Okay nice one thanks!

There are already some existing Viewport UI tests for these sort of things, might be worth checking they all still pass and seeing if we could potentially add a test for this behavior too.